pyne/alara.py

```diff
@@ -92,27 +95,23 @@ def mesh_to_fluxin(flux_mesh, flux_tag, fluxin="fluxin.out",
 
     # Establish for loop bounds based on if forward or backward printing
     # is requested
-    if not reverse:
-        start = 0
-        stop = num_e_groups
-        direction = 1
-    else:
-        start = num_e_groups - 1
-        stop = -1
-        direction = -1
-
+    output_list = []
     output = u""
     if not sub_voxel:
         for i, mat, ve in flux_mesh:
-            # print flux data to file
-            output = _output_flux(ve, tag_flux, output, start, stop, direction)
+            # append each block to list
+            output_list.append(_output_flux_block(ve, tag_flux, reverse))
+            if print_progress > 0 and i % print_progress == 0:
+                print(f"processing mesh element {i}")
     else:
         ves = list(flux_mesh.iter_ve())
-        for row in cell_fracs:
+        for i, row in enumerate(cell_fracs):
             if len(cell_mats[row['cell']].comp) != 0:
-                output = _output_flux(ves[row['idx']], tag_flux, output, start,
-                                      stop, direction)
+                output_list.append(_output_flux_block(ves[row['idx']], tag_flux, reverse))
+            if print_progress > 0 and i % print_progress == 0:
+                print(f"processing mesh element {row['idx']}")
 
+    output = ''.join([value for value in output_list])
     with open(fluxin, "w") as f:
         f.write(output)
```

Comment:
Would it maybe be even faster if the file was opened at the beginning of the method and each block was written as it is created, rather than appended to the `output_list`?

Reply:
I will make a test to see whether it is faster.

Reply:
Test results:
| num_ves | 4 | 100 | 1000 | 10000 |
|--- | --- | --- | --- | --- |
|join then write, time (s) | 1.0e-3 | 1.8e-2 | 0.14 | 1.46 |
| write block by block (s) | 1e-3 | 1.47e-2 | 0.149 | 1.44 |

Reply:
Write block by block implemented in https://github.com/pyne/pyne/pull/1397/commits/cfa164c87ecbadd7700948b6a2d9888645633fc4 and https://github.com/pyne/pyne/pull/1397/commits/1576ff0e3decfbf72806808b26dc504dc23d66f3.